game/atmosphere.rpy

```diff
@@ -0,0 +1,105 @@
+    _SKY_AND_DIM_MAP = {
+        "sky_day overcast" : "dim light",
+        "sky_day rain" : "dim medium",
+        "sky_day sunny" : None,
+        "sky_day thunder" : "dim heavy"
+    }
+
+    _MUFFLED_RAIN_PATH = "mod_assets/sfx/rain_muffled.mp3"
+
+    def show_random_sky():
+        """
+        Shows a randomised sky placeholder with associated dimming effect.
+        """
+
+        # Select the sky and dimming effect
+        sky, dim = random.choice(list(_SKY_AND_DIM_MAP.items()))
+
+        # Show the sky
+        renpy.show(name=sky, zorder=_SKY_Z_INDEX)
+        renpy.with_statement(trans=store.weather_change_transition)
+
+        # Add the dimming effect matching the sky, if it exists
+        if dim:
+            renpy.show(name=dim, zorder=_DIM_Z_INDEX)
+            renpy.with_statement(trans=store.dim_change_transition)
+
+        # Play rain sfx if the chosen weather is rainy
+        if sky in ("rain", "thunder"):
+            renpy.music.play(filenames=_MUFFLED_RAIN_PATH, channel="weather_loop", fadein=3.0)
+
+    def show_sky(weather_type):
+        """
+        Shows the specified sky placeholder with associated dimming effect.
+
+        IN:
+            weather_type - int; WEATHER_OVERCAST, WEATHER_RAIN, WEATHER_THUNDER or WEATHER_SUNNY
+        """
+        if (isinstance(weather_type, int)):
+            if weather_type == WEATHER_OVERCAST:
+                renpy.show(name="sky_day overcast", zorder=_SKY_Z_INDEX)
+                renpy.with_statement(trans=store.weather_change_transition)
+
+                renpy.show(name="dim light", zorder=_DIM_Z_INDEX)
+                renpy.with_statement(trans=store.dim_change_transition)
+
+                renpy.music.stop(channel="weather_loop", fadeout=5.0)
+
+            elif weather_type == WEATHER_RAIN:
+                renpy.show(name="sky_day rain", zorder=_SKY_Z_INDEX)
+                renpy.with_statement(trans=store.weather_change_transition)
+
+                renpy.show(name="dim medium", zorder=_DIM_Z_INDEX)
+                renpy.with_statement(trans=store.dim_change_transition)
+
+                renpy.music.play(filenames=_MUFFLED_RAIN_PATH, channel="weather_loop", fadein=3.0)
+
+            elif weather_type == WEATHER_THUNDER:
+                renpy.show(name="sky_day thunder", zorder=_SKY_Z_INDEX)
+                renpy.with_statement(trans=store.weather_change_transition)
+
+                renpy.show(name="dim heavy", zorder=_DIM_Z_INDEX)
+                renpy.with_statement(trans=store.dim_change_transition)
+
+                renpy.music.play(filenames=_MUFFLED_RAIN_PATH, channel="weather_loop", fadein=3.0)
+
+            else:
+                renpy.show(name="sky_day sunny", zorder=_SKY_Z_INDEX)
+                renpy.with_statement(trans=store.weather_change_transition)
+                renpy.hide(name="dim")
+
+                renpy.music.stop(channel="weather_loop", fadeout=5.0)
+
+        else:
```

Comment:
Most of the code here is similar. We should just use the if blocks to set variables on the things which change, then use common code to do the actions, this will make maintenance much simpler and the overall function much shorter

Reply:
Also adding a TODO for weather would be handy, we should aim to make them classes so they have all the information they need including dimming with references to the images for that too in them, that way they can be displayed more simply

Reply:
Got it